lib/statistics/client.rb

```diff
@@ -121,10 +121,11 @@ def initialize
       def fetch_events(group_id:, since_at: nil, until_at: nil)
         params = {
           fields: %i(attending_count start_time owner),
-          limit: 100,
-          since: since_at,
-          until: until_at
-        }.compact
+          limit: 100
+        }.tap do |h|
+          h[:since] = (since_at.utc + since_at.utc_offset).to_i if since_at
+          h[:until] = (until_at.utc + until_at.utc_offset).to_i if until_at
```

Comment:
月末のイベントが翌月に重複していたので、UTC時刻で範囲指定する必要があるようです。
ただ、リファレンスが見つからずちょっと確証がないです。

事象としては、
2013年6月30日(日) 13:30-14:30のCoderDojoTokyoのイベントが、6月集計分と7月集計分のAPIのレスポンスに含まれており、
7月集計分をinsertする際にユニーク制約でARがraiseしていました。
（9時間ズレなら15時なんですよねぇ。。。）

Reply:
`h[:since] = since_at.to_i + (3600 * 6) if since_at`
試しに1時間ずつ時刻をズラしていって6時間オフセットしたらエラーが出なくなった。
`2013/07/01 00:00:00+09:00`がsince_atに入っている時刻なので、
```
5.1.4@2.4.2 (main)> (Time.zone.local(2013,07) + 3600*6).in_time_zone('UTC')
=> Sun, 30 Jun 2013 21:00:00 UTC +00:00
```
と同義であり、UTCでも6/30の扱いのように見える。
なんだろうこれ、検索時のタイムゾーンはUTCでもないということなのか🤔

Reply:
https://developers.facebook.com/docs/graph-api/reference/v2.10/group/events

> By default this will only return events within the last two weeks, use until or since parameters to modify this range.

タイムゾーンについては特に書かれていなかった。

Reply:
一次情報がみつからなくてちょっと真偽がはっきりしないんですが以下の記事を見つけました。

https://trim.bz/2013/01/24/1799
> フェイスブックのイベントの取得をAPI介してやっていたところイベントの開催日時のタイムゾーンがよくわからない
てっきり標準時だと思っていたらちがう
調べてみたら太平洋時間（PST）というやつだった
これはUTC-8でフェイスブックの本社があるロサンゼルスのタイムゾーンだった
彼らはきっと自分の中心が世界の中心だと思ってるんでしょうね
といういやみはともかく、去年一度、標準時に変えたらしいんだけどまたもどしたとかよくわからない
とにかく現時点ではPSTです

ほほぅ🤔ってことで、
> h[:since] = since_at.to_i + (3600 * 6) if since_at
試しに1時間ずつ時刻をズラしていって6時間オフセットしたらエラーが出なくなった。

https://github.com/coderdojo-japan/coderdojo.jp/pull/182#discussion_r148932509 について検証してみました。

```
5.1.4@2.4.2 (main)> jst = ActiveSupport::TimeZone['Asia/Tokyo']
=> #<ActiveSupport::TimeZone:0x00007fdb55d5d788
 @name="Asia/Tokyo",
 @tzinfo=#<TZInfo::DataTimezone: Asia/Tokyo>,
 @utc_offset=nil>
5.1.4@2.4.2 (main)> (jst.local(2013,07) + 3600*5).in_time_zone('Pacific Time (US & Canada)')
=> Sun, 30 Jun 2013 13:00:00 PDT -07:00
5.1.4@2.4.2 (main)> (jst.local(2013,07) + 3600*6).in_time_zone('Pacific Time (US & Canada)')
=> Sun, 30 Jun 2013 14:00:00 PDT -07:00
```
「2013年6月30日(日) 13:30-14:30のCoderDojoTokyoのイベント」の開始が13:30であることから、5時間のオフセットだと検索に引っかかり、6時間のオフセットだと検索にひっかからなくなる、というのがこれで説明がつくことがわかりました。
つまり、フェイスブックのAPIがPST（RubyではPDT -7:00?）のタイムゾーンで時刻を渡してやる必要がある、ということのようです。
PDTとJSTのオフセットは8時間なので`3600*8`を足した時刻のUNIXタイムスタンプをパラメータに渡してやると良さそう。

Reply:
@yasulab @hanachin 
ここだけレビューをお願いしたいです。🙏
問題無さそうならマージしてしまって、 #183 も取り込むので #184 までいけそうだなと考えています😉

Reply:
コード確認しました! ✅ 👀 問題なさそうなので、マージしちゃいますね ;)

Reply:
ここ難しいですね

> 一次情報がみつからなくてちょっと真偽がはっきりしないんですが以下の記事を見つけました。

「site:facebook.com timezone api」みたいな感じで検索するとPSTでつかわれてるというFacebookヘルプチームからの解答がひっかかりました。

> Thanks for your question. Yes, We use UTC-8 Pacific Standard Time for all reporting. PDT and PST are interchangeable.
https://www.facebook.com/help/audiencenetwork/community/question/?id=1949627351935296

ということなので、オフセット足してよいと思います。